integration/smoke/run_smoke_parallel.sh

```diff
@@ -0,0 +1,213 @@
+#  SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+#  LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+#  DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+#  THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+#  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY LOG OF THE USE
+#  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+
+source smoke_env.sh
+
+# Variables to configure the set of test cases run by the main loop
+#
+# Applications with a valid 2-node configuration
+APPS_2NODE="dgemm_tiny dgemm nasft nekbone hpcg pennant hpl_mkl hpl_netlib"
+# Applications with no 2-node configuration
+APPS_1NODE="minife amg"
+# Experiment types to be run
+ALL_EXP="monitor power_sweep power_balancer_energy frequency_sweep barrier_frequency_sweep uncore_frequency_sweep"
+# Name of file to save results
+LOG_NAME="smoke_test_run_results.log"
+
+function pass_result {
+    # String to be inserted in a script to log a test case as
+    # completed with PASS result.
+    #
+    # Inputs:
+    #   $1: name of the application
+    #   $2: type of experiment
+    #   $3: Slurm job ID
+    #   LOG_NAME (global): log file to append results to
+    #
+    # Returns: a line to be inserted in a script to log the result
+
+    echo "echo \"${1} ${2} ${3} PASS\" >> ${LOG_NAME}"
+}
+
+function fail_result {
+    # String to be inserted in a script to log a test case as
+    # completed with FAIL result.
+    #
+    # Inputs:
+    #   $1: name of the application
+    #   $2: type of experiment
+    #   $3: Slurm job ID
+    #   LOG_NAME (global): log file to append results to
+    #
+    # Returns: a line to be inserted in a script to log the result
+
+    echo "echo \"${1} ${2} ${3} FAIL\" >> ${LOG_NAME}"
+}
+
+function skip_result {
+    # Logs a test case as skipped because the required run script is
+    # missing.
+    #
+    # Inputs:
+    #   $1: name of the application
+    #   $2: type of experiment
+    #   LOG_NAME (global): log file to append results to
+
+    echo "${1} ${2} SKIP" >> ${LOG_NAME}
+}
+
+function run_script_name {
+    # Generates the name of the run script for a given combination of
+    # experiment type and application.
+    #
+    # Inputs:
+    #   $1: name of the application
+    #   $2: type of experiment
+    #   GEOPM_SOURCE (global): root directory of the GEOPM source code
+    #
+    # Returns: name of the run script in integration/experiments that
+    # corresponds
+    local EXP_DIR=${2}
+    if [ "${2}" == "barrier_frequency_sweep" ] ||
+       [ "${2}" == "power_balancer_energy" ]; then
+        EXP_DIR="energy_efficiency"
+    fi
+    echo "${GEOPM_SOURCE}/integration/experiment/${EXP_DIR}/run_${2}_${1}.py"
+}
+
+function gen_sbatch {
+    # Creates a slurm batch script to run the given smoke test
+    # experiment.
+    #
+    # TODO: This function has some copied code from
```

Comment:
that script has some extra slurm options like account that might be useful to move here

Reply:
It would be nice if you could call that script here instead of copying a big chunk of the code.